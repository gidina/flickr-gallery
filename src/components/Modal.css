.gallery_item_full {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  background-color: #fff;
  background-color: rgba(255, 255, 255, 0.6);
  z-index: 100;
  overflow: hidden;
}

.gallery_item_full img {
  width: 100%;
}

.gallery_item_full embed {
  width: 100%;
  height: 70%;
}

.box {
  margin: auto;
  background: #fff;
  padding: 2em;
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  overflow: auto;
  width: 50%;
  height: 80%;
  text-align: left;
  border: 1px solid #dadada;
}

.box h3 {
  font-family: "Leckerli One", cursive;
  color: #f88484;
}

.close {
  cursor: pointer;
  position: absolute;
  top: 0.5em;
  right: 0.5em;
  color: #777;
  text-decoration: none;
  display: inline-block;
  font-size: 2em;
}

.close:hover {
  color: #f88484;
}
