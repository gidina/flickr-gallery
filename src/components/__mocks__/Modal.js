import React from "react";
export default class Modal extends React.Component {
    render() {
        return <modal-mock {...this.props}>{this.props.children}</modal-mock>;
    }
}