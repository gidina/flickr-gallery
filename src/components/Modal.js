import React, { Fragment } from "react";
import Loader from "./Loader";
import "./Modal.css";
import { blueColor } from "../config";

const Modal = ({ isLoading, photo, onClose }) => {
  return (
    <div className="gallery_item_full">
      {isLoading ? (
        <Loader color={blueColor} />
      ) : (
        <Fragment>
          <span className="close" onClick={() => onClose()}>
            &times;
          </span>
          <div className="box">
            {photo.isVideo ? (
              <embed src={photo.source} />
            ) : (
              <img src={photo.source} alt="" onClick={() => window.open(photo.redirectURL)} />
            )}
            <h3>{photo.title}</h3>
            <p>{photo.description}</p>
          </div>
        </Fragment>
      )}
    </div>
  );
};

{/* <img src={photo.source} alt="" onClick={() => window.location.href = photo.redirectURL} /> */}

// Modal.protoTypes = {
// onClose: PropTypes.function.isRequires
// };

export default Modal;
