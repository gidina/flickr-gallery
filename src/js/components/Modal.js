import React, { Component, Fragment } from 'react';
import './Modal.css';

const API_URL_GET_PHOTOS_SIZES = photoId => `https://api.flickr.com/services/rest/?method=flickr.photos.getSizes&api_key=1ffb89f5dcaa0f0c96c86fd180deade0&photo_id=${photoId}&format=json&nojsoncallback=1`;

class Modal extends Component {
  state = { visible: false }
  componentDidMount() {
    const { photoId, media } = this.props;

    fetch(API_URL_GET_PHOTOS_SIZES(photoId))
      .then(response => response.json())
    //   .then(myJson => console.log(myJson));

    // http://www.flickr.com/photos/{user-id|custom-url}/{photo-id}/play/{site|mobile|hd|orig}/{secret|originalsecret}/


      .then(myJson => this.setState({...this.state, [media]: myJson.sizes.size[media === "video" ? 10 : 1].source}));
  }
  render() {
    const { photoId, media } = this.props;
    const { visible } = this.state;

    console.log(this.state.video);
    return (
        <Fragment>
            <button type="button" onClick={() => this.setState({ visible: true })}>Click to view</button>
            <div className="modal" style={{display: visible ? "block" : "none"}}>
                <span className="close" onClick={() => this.setState({ visible: false })}>&times;</span>
                {media === "video" ? <video width="320" height="240" controls>
      <source src={this.state.video} type="video/mp4" />
      Your browser does not support the video tag.
    </video> : <img className="modal-content" src={this.state.photo} alt="TODO: cal passar ALT" />}
                <div id="caption"></div>
            </div>
        </Fragment>
    );
  }
}

export default Modal;