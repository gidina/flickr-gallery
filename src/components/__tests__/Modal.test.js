import React from "react";
import Modal from "../Modal";
import renderer from "react-test-renderer";

jest.mock("../Loader", () => "LoaderMock");

test("Renders blue (#5690F7) Loader component if isLoading property is true", () => {
    const tree = renderer
        .create(<Modal isLoading={true} photo={{}} onClose={() => {}}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe("Renders photo properties if isLoading property is false", () => {
    test("Renders 'embed' with properly source if isVideo property is true", () => {
        const photoTest = {
            isVideo: true,
            source: "/source",
            title: "Photo 1",
            redirectURL: "/redirect-url",
            username: "username",
            description: "Description Photo 1",
        };
        const tree = renderer
            .create(<Modal isLoading={false} photo={photoTest} onClose={() => {}}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    test("Renders 'img' with properly source if isVideo property is false", () => {
        const photoTest = {
            isVideo: false,
            source: "/source",
            title: "Photo 1",
            redirectURL: "/redirect-url",
            username: "username",
            description: "Description Photo 1",
        };
        const tree = renderer
            .create(<Modal isLoading={false} photo={photoTest} onClose={() => {}}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});