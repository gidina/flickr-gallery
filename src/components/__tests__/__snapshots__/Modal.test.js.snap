// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders blue (#5690F7) Loader component if isLoading property is true 1`] = `
<div
  className="gallery_item_full"
>
  <LoaderMock
    color="#5690F7"
  />
</div>
`;

exports[`Renders photo properties if isLoading property is false Renders 'embed' with properly source if isVideo property is true 1`] = `
<div
  className="gallery_item_full"
>
  <span
    className="close"
    onClick={[Function]}
  >
    ×
  </span>
  <div
    className="box"
  >
    <embed
      src="/source"
    />
    <h3>
      Photo 1
    </h3>
     - 
    <h4
      onClick={[Function]}
    >
      username
    </h4>
    <p>
      Description Photo 1
    </p>
  </div>
</div>
`;

exports[`Renders photo properties if isLoading property is false Renders 'img' with properly source if isVideo property is false 1`] = `
<div
  className="gallery_item_full"
>
  <span
    className="close"
    onClick={[Function]}
  >
    ×
  </span>
  <div
    className="box"
  >
    <img
      alt=""
      src="/source"
    />
    <h3>
      Photo 1
    </h3>
     - 
    <h4
      onClick={[Function]}
    >
      username
    </h4>
    <p>
      Description Photo 1
    </p>
  </div>
</div>
`;
